crates/amaru-stores/src/rocksdb/consensus.rs

```diff
@@ -225,10 +191,13 @@ macro_rules! impl_ReadOnlyChainStore {
             fn get_children(&self, hash: &Hash<32>) -> Vec<Hash<32>> {
                 let mut result = Vec::new();
                 let prefix = [&CHILD_PREFIX[..], &hash[..]].concat();
-                let cf = self.db.cf_handle(CHILDREN_CF).expect("missing children CF");
-                for res in self.db.prefix_iterator_cf(&cf, &prefix) {
+                for res in self.db.prefix_iterator(&prefix) {
                     match res {
                         Ok((key, _value)) => {
+                            // Stop once we get keys that don't start with the required prefix
+                            if !key.starts_with(&prefix) {
```

Comment:
It seems to me from casual inspection of db_iterator.rs as well as a grok answer that this check shouldn’t be necessary. Do you have evidence to the contrary? Then we’d need to audit all RocksDB usage.

Reply:
The problem see to occur because we don't set any prefix extractor.

This solves the issue and allows to get rid of this line:
```
        opts.set_prefix_extractor(rocksdb::SliceTransform::create_fixed_prefix(
            CONSENSUS_PREFIX_LEN,
        ));
```